### ISSUE TITLE
Does ReflectionAnalyser attempt to run PHP code it comes across?

### ISSUE THREAD

**Issue by range-of-motion** [zircote/swagger-php#1124]
Hi 👋 

I'm upgrading a project that's using a pre-v3.0.0 commit of this package (yes, I know). I'm used to this package analysing PHP files, and generating an OpenAPI document based on the PHPDoc it found.

I'm trying to upgrade to v4.2 and it looks like this package is now trying to actually run the PHP files it comes across. I'm under the impression that this has to do with the move to `ReflectionAnalyser` (we would have probably used `TokenAnalyser` before). 

Am I making the right assumption here? 

Thanks,
Daniël

**Comment by DerManoMann** [comment]
Yes, that is correct.

In order to use reflection the code needs to be loaded; the assumption is that the code follows good practices where each file contains a single class and can be loaded side-effect free.
I suppose there are a couple of improvements that could be made but even if swagger-php would not explicitely load a file the autoloader would (should).

You can use version 4 and the `TokenAnalyser` by adding `--legacy` to the command line. That should give you the same behaviour as version 3 but with most, if not all, improvements made in version 4.

**Comment by DerManoMann** [comment]
Had a quick stab at limiting loading of files - turns out all files processed are loaded right now, whether they contain classes (etc) or not.

Here is a PR to limit that: https://github.com/zircote/swagger-php/pull/1126

Let me know if that helps.

**Comment by range-of-motion** [comment]
Thanks for confirming that 🙂 

If I want to use the `ReflectionAnalyser` it even runs commands (using the Symfony package) or tests (using PHPUnit) for example. I'll give you an example of my project structure.

```
models/
├─ User.php
├─ Role.php
├─ Permission.php
Commands/
├─ RunSomething.php
tests/
├─ ExampleTest.php
bootstrap.php
```

Is there a reason why my commands and tests are executing? They are separate classes with individual functions, and shouldn't trigger upon instantiation. Am I doing something wrong?

**Comment by DerManoMann** [comment]
Depends on what you specify as directory to process - if `Commands` is included to be scanned then, yes, that is what is happening atm.

You could try to be more specific, perhaps only scan models and controllers or similar. The command line does accept multiple directories/files.

**Comment by range-of-motion** [comment]
Got it, I'll try tomorrow. Thanks for the quick answers! 👏

**Comment by DerManoMann** [comment]
No problem. Thanks, actually! The question made me realise that the `require_once` is in there just because some fo the examples hadn't been migrated to proper autoloading, so this was a good reason to do some cleanup :)

Next version will be without loading files at all! Still, good to limit what folders you are processing to keep things snappy.

**Comment by DerManoMann** [comment]
Fixed in 4.2.9
